propagation.go

```diff
@@ -71,7 +77,7 @@ func newTextMapPropagator(headerKeys *HeadersConfig, metrics Metrics) *textMapPr
 	}
 }
 
-func newHTTPHeaderPropagator(headerKeys *HeadersConfig, metrics Metrics) *textMapPropagator {
+func NewHTTPHeaderPropagator(headerKeys *HeadersConfig, metrics Metrics) *textMapPropagator {
```

Comment:
exported func NewHTTPHeaderPropagator returns unexported type *jaeger.textMapPropagator, which can be annoying to use